File: U5/Java/U5P_7.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package com.mycompan.u5p_7;
import java.util.Scanner;
/**
 *
 * @author alfre
 */
public class U5P_7 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Solicitar al usuario que ingrese el precio y la cantidad de productos
        System.out.print("Ingrese el precio unitario del producto: ");
        double precioUnitario = scanner.nextDouble();

        System.out.print("Ingrese la cantidad de productos: ");
        int cantidad = scanner.nextInt();

        //  Subtotal
        double subtotal = calcularSubtotal(precioUnitario, cantidad);

        // IVA
        double iva = calcularIVA(subtotal);

        // Total
        double total = calcularTotal(subtotal, iva);

        // resultados
        System.out.println("\nDetalles de la Factura:");
        System.out.println("Precio Unitario: $" + precioUnitario);
        System.out.println("Cantidad: " + cantidad);
        System.out.println("Subtotal: $" + subtotal);
        System.out.println("IVA (16%): $" + iva);
        System.out.println("Total: $" + total);
    }

    // Subtotal
    private static double calcularSubtotal(double precioUnitario, int cantidad) {
        return precioUnitario * cantidad;
    }

    //  IVA (16%)
    private static double calcularIVA(double subtotal) {
        final double PORCENTAJE_IVA = 0.16;
        return subtotal * PORCENTAJE_IVA;
    }

    // Función para calcular el Total
    private static double calcularTotal(double subtotal, double iva) {
        return subtotal + iva;
    }
}

File: U5/C#/U5P_7.cs
using System;

namespace U5P_7
{
    class Program
    {
        static void Main(string[] args)
        {
            // Solicitar al usuario que ingrese el precio y la cantidad de productos
            Console.Write("Ingrese el precio unitario del producto: ");
            double precioUnitario = Convert.ToDouble(Console.ReadLine());

            Console.Write("Ingrese la cantidad de productos: ");
            int cantidad = Convert.ToInt32(Console.ReadLine());

            // Subtotal
            double subtotal = CalcularSubtotal(precioUnitario, cantidad);

            // IVA
            double iva = CalcularIVA(subtotal);

            // Total
            double total = CalcularTotal(subtotal, iva);

            // Resultados
            Console.WriteLine("\nDetalles de la Factura:");
            Console.WriteLine("Precio Unitario: $" + precioUnitario);
            Console.WriteLine("Cantidad: " + cantidad);
            Console.WriteLine("Subtotal: $" + subtotal);
            Console.WriteLine("IVA (16%): $" + iva);
            Console.WriteLine("Total: $" + total);
        }

        // Subtotal
        private static double CalcularSubtotal(double precioUnitario, int cantidad)
        {
            return precioUnitario * cantidad;
        }

        // IVA (16%)
        private static double CalcularIVA(double subtotal)
        {
            const double PORCENTAJE_IVA = 0.16;
            return subtotal * PORCENTAJE_IVA;
        }

        // Función para calcular el Total
        private static double CalcularTotal(double subtotal, double iva)
        {
            return subtotal + iva;
        }
    }
}
